docs/containerization.md

```diff
@@ -1,66 +1,100 @@
+nvidia-ctk cdi list
 ```
 
-Or image: TBD (am I allowed to have a public image with references to lab in it?)
+Example output:
 
-This `Containerfile` is based on Nvidia CUDA image, which lucky for us plugs
-directly into Podman via their `nvidia-container-toolkit`! The `ubi9` base
-image does not have most packages installed. The bulk of the `Containerfile` is
-spent configuring your system so `ilab` can be installed and run properly.
-`ubi9` as compared to `ubuntu` cannot install the entire `nvidia-12-4` toolkit.
-This did not impact performance during testing.
+```shell
+INFO[0000] Found 2 CDI devices
+nvidia.com/gpu=0
+nvidia.com/gpu=all
+```
+
+Finally, you can generate the CDI configuration for the NVIDIA devices:
 
 ```shell
-1. podman build –ssh=default -f <Containerfile_Path>
-2. curl -s -L https://nvidia.github.io/libnvidia-container/stable/rpm/nvidia-container-toolkit.repo |   sudo tee /etc/yum.repos.d/nvidia-container-toolkit.repo
-3. sudo yum-config-manager --enable nvidia-container-toolkit-experimental
-4. sudo dnf install -y nvidia-container-toolkit
-5. sudo nvidia-ctk cdi generate --output=/etc/cdi/nvidia.yaml
-6. nvidia-ctk cdi list
-    Example output: 
-    INFO[0000] Found 2 CDI devices
-    nvidia.com/gpu=0
-    nvidia.com/gpu=all
-7. podman run --device nvidia.com/gpu=0  --security-opt=label=disable -it <IMAGE_ID>
+sudo nvidia-ctk cdi generate --output=/etc/cdi/nvidia.yaml
 ```
 
-Voila! You now have a container with CUDA and GPUs enabled!
+### Run the CUDA-accelerated InstructLab container
 
-### Sources
+When running our model, we want to create some paths that will be mounted in
+the container to provide data persistence. As an unprivileged user, you will
+run a rootless container, so you need to let the internal user (UID 1001)
+access the files in the host. For that, we use `podman unshare chown`. The
+directory will contain InstructLab's configuration, models, taxonomy,
+and Hugging Face download cache.
 
-[Nvidia Container Toolkit Install Guide](https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/latest/install-guide.html)
+```shell
+mkdir -p $HOME/.config/instructlab
+podman unshare chown 1001:1001 -R $HOME/.config/instructlab
+```
 
-[Podman Support for Container Device Interface](https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/latest/cdi-support.html)
+Then, we can run the container, mounting the above folder and using the first
+NVIDIA GPU.
 
-### Notes
+```shell
+podman run --rm -it --user 1001 --device nvidia.com/gpu=0 --volume $HOME/.config/instructlab:/opt/app-root/src:z localhost/instructlab:cuda
+```
+
+The above command will print the help, as we didn't pass any argument.
 
-Thanks to Taj Salawu for figuring out how to pass the git ssh keys properly!
+Let's initialize our configuration, download the model and start the chatbot.
+
+```shell
+podman run --rm -it --device nvidia.com/gpu=0 --volume $HOME/.config/instructlab:/opt/app-root/src:z localhost/instructlab:cuda init
```

Comment:
I don't know if I like this flow, I think we should have a way as well for users to run all of these in a ctr, rather than passing in the command they want to run.

Maybe we can have a different build target for this?

Reply:
It's coming from Fabien as well.

For my ROCm container, I suggest that the user enters the container and uses it as an interactive shell.